### PR TITLE
Issue126: upgrade go version to  go1.21

### PR DESCRIPTION
**Pull request by ping-ke**
issue 126: https://github.com/ethstorage/es-node/issues/126

How to test
install go1.21 and build es-node using go1.21, and run es-node successfully.
```
go install golang.org/dl/go1.21.4@latest
go1.21.4 download
cd cmd/es-node
go1.21.4 build
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/ethstorage/go-ethstorage
 
-go 1.20
+go 1.21
```
Comment (qzhodl) [inline go.mod:3]: We may need to update the guide [here](https://github.com/ethstorage/es-node/blob/main/GUIDE.md#system-environment)

Comment (qzhodl) [reply go.mod:3]: Also adding @syntrust here to check if there are other places to update

**Review by syntrust — COMMENTED** [review]
Dockerfile may need to be updated too.

File: GUIDE.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ Please refer to [this section](/README.md/#minimum-hardware-requirements) for ha
 ## System Environment
  - Ubuntu 20.04+ (tested and verified)
  - (Optional) Docker 24.0.5+ (would simplify the process)
- - (Optional) Go 1.20.* (not compatible with Go 1.21 yet) and Node.js 16+ (can be installed following the [steps](#install-dependencies))
+ - (Optional) Go 1.21.* and Node.js 16+ (can be installed following the [steps](#install-dependencies))
```
Comment (qzhodl) [inline GUIDE.md:22]: (Optional) Go 1.20+ and Node.js 16+ (can be installed following the [steps](#install-dependencies))

**Review by qzhodl — APPROVED** [review]

**Review by syntrust — APPROVED** [review]

**Pull request merged**